lib/decidim/internal_evaluation/export.rb

```diff
@@ -12,20 +12,46 @@ module Export
       #
       # Returns an Arel::Relation with all the internal evaluations for that component and resource.
       def internal_evaluations_for_resource(resource_class, component, user)
+        filtered_collection(resource_class, component, user).map do |proposal|
+          proposal.valuation_assignments.map do |assignment|
+            author = assignment.valuator_role&.user
+
+            next if author.blank?
+
+            InternalEvaluation.find_or_initialize_by(proposal:, author:)
+          end
+        end.flatten.compact
+      end
+
+      # Internal: Returns the filtered collection for the given resource class, component and user.
+      #
+      # resource_class - The resource's Class
+      # component      - The component where the resource is scoped to.
+      # user           - The user that is requesting the export.
+      #
+      # Returns an Arel::Relation with the filtered collection.
+      def filtered_collection(resource_class, component, user)
```

Comment:
I think something like this could be cleaner and better aligned with the format we are used to seeing in Decidim:

```ruby
      def filtered_collection(resource_class, component, user)
        @filtered_collection ||= begin
          collection = resource_class
                       .not_hidden
                       .published
                       .where(component:)
                       .where.not(valuation_assignments_count: 0)

          participatory_space = component.participatory_space
          user_is_valuator = participatory_space.user_roles(:valuator).where(user:).any?

          if user_is_valuator
            collection.with_valuation_assigned_to(user, participatory_space)
          else
            collection
          end
        end
      end
```

Reply:
Done ✅